react-native/src/index.js

```diff
@@ -671,10 +671,9 @@ export class AbstractSession extends Session {
   /**
    * Cancels running the session.
    */
-  cancel() {
-    if (sessionId === undefined) {
-      return FFmpegKitReactNativeModule.cancel();
-    } else {
+  async cancel() {
```

Comment:
How do you plan to use the `promise` returned by this method? Unfortunately, we cannot guarantee that session is cancelled when the promise returned by `FFmpegKitReactNativeModule.cancelSession()` is fulfilled. `cancel()` only starts the cancellation.

Reply:
Shouldn't `cancel()` always return a promise?

https://github.com/tanersener/ffmpeg-kit/blob/c63401acf34e9d4e218340b496729bab987815f8/react-native/src/index.d.ts#L61

Also it would would be a problem if someone does `cancel().then()` instead of `await cancel()` if there is no `sessionId`.

We can remove returning a promise all together if you prefer but then we wouldn't be able to catch errors from `FFmpegKitReactNativeModule.cancelSession()`.

What about adding a comment?
```
/**
 * Cancels running the session. Only starts cancellation. Does not guarantee that session is cancelled when promise resolves. 
 */
  async cancel() {
```

Reply:
Yeah, updated comment looks good. How about returning a Promise error in case `sessionId` is undefined. Something like `Promise.reject`. Would that address the issues?